deadd-notification-center: 2022-04-20 -> 2022-11-07 and build from source

S-NA opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Update and switch to building from source. 

cc @Pacman99 @teto for extra testing if possible.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Pacman99 commented:
I actually switched to GNOME a while back and haven't had need for this package, if your willing to maintain it, could you remove me as maintainer. Thanks for working on it!

aanderse commented on pkgs/applications/misc/deadd-notification-center/default.nix:
```diff
@@ -1,62 +1,61 @@
+{ mkDerivation, haskellPackages, fetchFromGitHub, lib, writeText }:
+
+let
+  systemd-service
```
It looks like upstream already provides a `systemd` user service: https://github.com/phuhl/linux_notification_center/blob/master/deadd-notification-center.service.in

You could use that instead.

S-NA committed:
deadd-notification-center: 2022-04-20 -> unstable-2022-11-07

Update deadd-notification-center to the latest revision. Change the
expression from using the prebuilt binary provided in the repository to
building from source instead as well as provide a functioning systemd
service file as the one provided by upstream is broken.

Remove pacman99 from and add Melkor333 to maintainers as requested.

S-NA commented:
@wineee I see you last updated the package. Mind reviewing this PR?

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/1462

SuperSandro2000 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -28572,7 +28572,9 @@ with pkgs;
   linssid = libsForQt5.callPackage ../applications/networking/linssid { };
 
-  deadd-notification-center = callPackage ../applications/misc/deadd-notification-center { };
+  deadd-notification-center
+    = haskell.lib.compose.justStaticExecutables
+        (haskellPackages.callPackage ../applications/misc/deadd-notification-center { });
```
```suggestion
  deadd-notification-center = haskell.lib.compose.justStaticExecutables (haskellPackages.callPackage ../applications/misc/deadd-notification-center { });
```

SuperSandro2000 commented on pkgs/applications/misc/deadd-notification-center/default.nix:
```diff
@@ -1,62 +1,61 @@
+  maintainers = with lib.maintainers; [ melkor333 sna ];
+  platforms = lib.platforms.linux;
 }
+
```
Thats one final new line to much

SuperSandro2000 commented on pkgs/applications/misc/deadd-notification-center/default.nix:
```diff
@@ -1,62 +1,61 @@
+{ mkDerivation, haskellPackages, fetchFromGitHub, lib, writeText }:
+
+let
+  systemd-service
+    = writeText "deadd-notification-center.service" ''
+        [Unit]
+        Description=Deadd Notification Center
+        PartOf=graphical-session.target
+
+        [Service]
+        Type=dbus
+        BusName=org.freedesktop.Notifications
+        ExecStart=$out/bin/deadd-notification-center
+
+        [Install]
+        WantedBy=graphical-session.target
+      '';
```
We could just echo that into the final location because creating a derivation and then cp'ing it is no necessary.

SuperSandro2000 commented on pkgs/applications/misc/deadd-notification-center/default.nix:
```diff
@@ -1,62 +1,61 @@
+mkDerivation rec {
   pname = "deadd-notification-center";
-  version = "2022-04-20";
+  version = "2022-11-07";
```
```suggestion
  version = "unstable-2022-11-07";
```

SuperSandro2000 commented on pkgs/applications/misc/deadd-notification-center/default.nix:
```diff
@@ -1,62 +1,61 @@
+        [Service]
+        Type=dbus
+        BusName=org.freedesktop.Notifications
+        ExecStart=$out/bin/deadd-notification-center
```
Then this workaround would also disappear.

Melkor333 commented:
@SuperSandro2000 Can you maybe review it again? From what I see @S-NA incorporated all your suggestions 🙂

Icy-Thought commented:
If you are still interested in the PR, it would be very helpful if you would push the PR to latest commit because of the recent yaml support and updates. :)

SuperSandro2000 reviewed (approved):
Please rebase

S-NA commented:
I've rebased my old update, I am going to refrain from updating to the latest upstream commit because when running the latest I notice:
```
notificationDaemon started
deadd-notification-center: /etc/xdg/deadd/deadd.css: openFile: does not exist (No such file or directory)
```
It does not properly fall back to searching the user's local config folder. This PR still provides building from source and a newer version than what is currently packaged which is preferable.

I will update to the latest commit when I get the time to look into why the fall back of the user's local config is not working. (In a different PR is that is OK with all.)
